qualifier-jeremy.py

```diff
@@ -0,0 +1,92 @@
+import datetime
+import re
+import typing
+
+
+class ArticleField:
+    def __init__(self, field_type: typing.Type[typing.Any]):
+        self.field_type = field_type
+
+    def __set_name__(self, owner, name):
+        self.name = name
+
+    def __get__(self, obj, type=None) -> typing.Type[typing.Any]:
+        return obj.__dict__.get(self.name)
+
+    def __set__(self, obj, value) -> None:
+        if not isinstance(value, self.field_type):
+            err = (f"expected an instance of type "
+                   f"'{self.field_type.__name__}' "
+                   f"for attribute {repr(self.name)}, "
+                   f"got '{type(value).__name__}' instead")
+            raise TypeError(err)
+
+        obj.__dict__[self.name] = value
+
+        if self.name == "content":
```

Comment:
this logic is way better placed as a property of the article class. Dealing with special cases in this generic descriptor should be avoided